Flutter app not building on Linux. CMake Error at flutter/ephemeral/.plugin_symlinks/realm/linux/CMakeLists.txt:38 (file): file does not recognize sub-command REAL_PATH

### What happened?

Flutter application is not building on Linux after installing realm package.

It looks like we're trying to use a [sub-command that has added in CMake 3.19](https://cmake.org/cmake/help/latest/command/file.html#real-path) in [CMakeLists.txt](https://github.com/realm/realm-dart/blob/ada7d5531814204be6497df97d9c330c0aa8af71/flutter/realm_flutter/linux/CMakeLists.txt), while specifying the minimum CMake version as 3.10.

Note: 
I installed Flutter using snap,
```
sudo snap install flutter --classic
```
which comes with CMake 3.16.3.

### Repro steps

Create an app:
```
flutter create my_app
cd my_app
```

Install realm:
```
flutter pub add realm
```

Try to launch the app:
```
flutter run -d linux
```

Output:
```
Launching lib/main.dart on Linux in debug mode...
CMake Error at flutter/ephemeral/.plugin_symlinks/realm/linux/CMakeLists.txt:38 (file):
  file does not recognize sub-command REAL_PATH


Could not find file `realm`
Could not find file `realm`
Building Linux application...
Exception: Unable to generate build files
```

### Version

Dart: 3.0.6 / Flutter: 3.10.6

### What Atlas Services are you using?

Local Database only

### What type of application is this?

Flutter Application

### Client OS and version

Ubuntu 22.04

### Code snippets

_No response_

### Stacktrace of the exception/crash you're getting

_No response_

### Relevant log output

```shell
flutter doctor -v

[✓] Flutter (Channel stable, 3.10.6, on Ubuntu 22.04.3 LTS 6.2.0-26-generic,
    locale en_US.UTF-8)
    • Flutter version 3.10.6 on channel stable at
      /home/thiago/snap/flutter/common/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f468f3366c (5 weeks ago), 2023-07-12 15:19:05 -0700
    • Engine revision cdbeda788a
    • Dart version 3.0.6
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /home/thiago/Android/Sdk
    • Platform android-33, build-tools 33.0.1
    • Java binary at: /usr/local/android-studio/jbr/bin/java
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 10.0.0-4ubuntu1
    • cmake version 3.16.3
    • ninja version 1.10.0
    • pkg-config version 0.29.1

[✓] Android Studio (version 2022.2)
    • Android Studio at /usr/local/android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code
    • VS Code at /snap/code/current
    • Flutter extension version 3.70.0

[✓] Connected device (2 available)
    • Linux (desktop) • linux  • linux-x64      • Ubuntu 22.04.3 LTS
      6.2.0-26-generic
    • Chrome (web)    • chrome • web-javascript • Google Chrome 116.0.5845.96

[✓] Network resources
    • All expected network resources are available.

• No issues found!
```

same prolem happens with me.

Ubuntu 22.04.3 LTS
64-bit
Flutter 3.13.2
Dart SDK version: 3.1.0

I updated cMake to:
cmake version 3.27.4

still getting the same error

CMake Error at flutter/ephemeral/.plugin_symlinks/realm/linux/CMakeLists.txt:38 (file):
  file does not recognize sub-command REAL_PATH
Could not find file `realm`
Exception: Unable to generate build files

In case the context is lost here is the output from trying again.


```
flutter run -d linux
Launching lib/main.dart on Linux in debug mode...
Could not find file 'realm'
Could not find file 'realm'
CMake Error at cmake_install.cmake:124 (file):
  file INSTALL cannot find
  "~/<projectroot>/linux/flutter/ephemeral/.plugin_symlinks/realm/linux/binary/linux/librealm_dart.so":
  No such file or directory.


Building Linux application...
Exception: Build process failed
```


I can confirm again that the issue still persists.
 - Ubuntu 23.10
 - Flutter 3.16.5
 - CMake 3.27.4
 - Realm 1.6.1
 
 The same app builds fine on Windows 11.

--- a/flutter/realm_flutter/linux/CMakeLists.txt
+++ b/flutter/realm_flutter/linux/CMakeLists.txt
@@ -33,6 +33,9 @@ include(${EPHEMERAL_DIR}/generated_config.cmake)
 
 set(APP_DIR "${CMAKE_CURRENT_SOURCE_DIR}/../../../../../..")
 
+file(REAL_PATH ${APP_DIR} ABSOLUTE_PATH_APP_DIR)
+# message ("ABSOLUTE_PATH_APP_DIR is ${ABSOLUTE_PATH_APP_DIR}")
+
 # message ("APP_DIR is ${APP_DIR}")
 set(APP_PUBSPEC_FILE "${APP_DIR}/pubspec.yaml")
 
@@ -53,6 +56,7 @@ add_definitions(-DBUNDLE_ID="${BUNDLE_ID}")
 # message ("FLUTTER_TOOL_ENVIRONMENT is ${FLUTTER_TOOL_ENVIRONMENT}")
 # message ("FLUTTER_ROOT is ${FLUTTER_ROOT}")
 execute_process(COMMAND "${FLUTTER_ROOT}/bin/dart" "run" "realm" "install" "--target-os-type" "linux" "--flavor" "flutter" # "--debug"
+  WORKING_DIRECTORY ${ABSOLUTE_PATH_APP_DIR}
   OUTPUT_VARIABLE output
   RESULT_VARIABLE result
 
@@ -64,7 +68,7 @@ message(STATUS "cmd result: ${result}")
 # message("CMAKE_HOST_SYSTEM_VERSION ${CMAKE_HOST_SYSTEM_VERSION}")
 execute_process(
   COMMAND "${FLUTTER_ROOT}/bin/dart" "run" "realm" "metrics" "--verbose" "--flutter-root" "${FLUTTER_ROOT}/bin" "--target-os-type" "linux" "--target-os-version" "${CMAKE_HOST_SYSTEM_VERSION}" # "--pause-isolates-on-start" "--enable-vm-service"
-
+  WORKING_DIRECTORY ${ABSOLUTE_PATH_APP_DIR}
   # COMMAND ${CMAKE_COMMAND} -E true
   OUTPUT_VARIABLE output
   RESULT_VARIABLE result

